### PR TITLE
[azure-core] fixes sphinx paths and some docstring formatting

### PR DESCRIPTION
**Pull request by kristapratico**

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: sdk/core/azure-core/azure/core/pipeline/policies/redirect.py [hunk 1]
```diff
@@ -48,11 +48,12 @@ class RedirectPolicy(HTTPPolicy):
     A redirect policy in the pipeline can be configured directly or per operation.
 
     Keyword arguments:
+
     :param redirects_allow: Whether the client allows redirects. Defaults to True.
     :param redirect_max: The maximum allowed redirects. Defaults to 30.
```
Comment (xiangyan99) [inline sdk/core/azure-core/azure/core/pipeline/policies/redirect.py]: Need to update as well?

File: sdk/core/azure-core/azure/core/pipeline/policies/redirect_async.py [hunk 2]
```diff
@@ -36,11 +36,12 @@ class AsyncRedirectPolicy(RedirectPolicy, AsyncHTTPPolicy):  # type: ignore
     An async redirect policy in the pipeline can be configured directly or per operation.
 
     Keyword arguments:
+
     :param redirects_allow: Whether the client allows redirects. Defaults to True.
     :param redirect_max: The maximum allowed redirects. Defaults to 30.
```
Comment (xiangyan99) [inline sdk/core/azure-core/azure/core/pipeline/policies/redirect_async.py]: Same

File: sdk/core/azure-core/azure/core/pipeline/policies/retry.py [hunk 3]
```diff
@@ -51,6 +51,7 @@ class RetryPolicy(HTTPPolicy):
     The retry policy in the pipeline can be configured directly, or tweaked on a per-call basis.
 
     Keyword arguments:
+
     :param int retry_total: Total number of retries to allow. Takes precedence over other counts.
      Default value is 10.
     :param int retry_connect: How many connection-related errors to retry on.
```
Comment (xiangyan99) [inline sdk/core/azure-core/azure/core/pipeline/policies/retry.py]: Same

File: sdk/core/azure-core/azure/core/pipeline/policies/retry_async.py [hunk 4]
```diff
@@ -45,6 +45,7 @@ class AsyncRetryPolicy(RetryPolicy, AsyncHTTPPolicy):  # type: ignore
     The async retry policy in the pipeline can be configured directly, or tweaked on a per-call basis.
 
     Keyword arguments:
+
     :param int retry_total: Total number of retries to allow. Takes precedence over other counts.
      Default value is 10.
     :param int retry_connect: How many connection-related errors to retry on.
```
Comment (xiangyan99) [inline sdk/core/azure-core/azure/core/pipeline/policies/retry_async.py]: +1

**Pull request merged**